src/sidebar/limits.js

```diff
@@ -0,0 +1,225 @@
+                                {submitting
+                                    ? i18n.t('Saving...')
+                                    : i18n.t('Save limits')}
+                            </Button>
+                            <Button
+                                small
+                                secondary
+                                onClick={() => setEditing(false)}
+                            >
+                                {i18n.t('Cancel')}
+                            </Button>
+                            {deleteLimitsButton}
+                        </ButtonStrip>
+                    </form>
+                )}
+            </ReactFinalForm.Form>
+        )
+    }
+
+    return (
+        <>
+            {averageValueInfo}
+            {min === null && max === null ? (
+                <>
+                    <div className={styles.placeholder}>
+                        {i18n.t('No limits set for this data item.')}
+                    </div>
+                    <Button small secondary onClick={() => setEditing(true)}>
+                        {i18n.t('Add limits')}
+                    </Button>
+                </>
+            ) : (
+                <>
+                    <div className={styles.limits}>
+                        {min !== null && (
+                            <div className={styles.limit}>
+                                <span className={styles.limitLabel}>
+                                    {i18n.t('Minimum')}
+                                </span>
+                                <span className={styles.limitValue}>{min}</span>
+                            </div>
+                        )}
+                        {min !== null && max !== null && (
+                            <div className={styles.spaceBetween}></div>
+                        )}
+                        {max !== null && (
+                            <div className={styles.limit}>
+                                <span className={styles.limitLabel}>
+                                    {i18n.t('Maximum')}
+                                </span>
+                                <span className={styles.limitValue}>{max}</span>
+                            </div>
+                        )}
+                    </div>
+                    <ButtonStrip>
+                        <Button
+                            small
+                            secondary
+                            onClick={() => setEditing(true)}
+                        >
+                            {i18n.t('Edit limits')}
+                        </Button>
+                        {deleteLimitsButton}
+                    </ButtonStrip>
+                </>
+            )}
+        </>
+    )
+}
+
+Limits.propTypes = {
+    itemId: PropTypes.string.isRequired,
+}
+
+const LimitsUnit = ({ itemId, itemType }) => (
+    <ToggleableUnit
+        title={i18n.t('Minimum and maximum limits')}
+        disabled={itemType !== 'numerical'}
```

Comment:
`numerical` isn't actually a valid item type: see https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java for list of possible types